### ISSUE TITLE
Error when publishing in terminal

### ISSUE THREAD

**Issue by dooboolab** [expo/expo#1355]
![image](https://user-images.githubusercontent.com/27461460/35526437-b85b79b4-056a-11e8-81d9-962c04bde90e.png)

It is ok when I publish with xde but always not working in a terminal. I remember someone wrote issue on this but still not solved and I can't find it.

Please make it work using terminal too for non xde users.

**Comment by brentvatne** [comment]
hi there. it does work in terminal for most people but something appears to be going wrong on your particular machine ;) could you share your project with us, or any more information about it? or also if you run it with EXPO_DEBUG=true like suggested to view the stack trace and let us see that, it would be useful. i'm sure you can appreciate that based on what you have provided it is impossible for us to help debug your issue, so give us all that you've got! :D

**Comment by dooboolab** [comment]
@brentvatne 
Thanks. I've just did what you wrote me and I have no idea what this is.
```
[exp] Uploading JavaScript bundles
[exp] There was an error publishing your project. Please check for any warnings.
[exp] Error: There was an error publishing your project. Please check for any warnings.
    at Object.<anonymous> (/xdl/src/Project.js:389:11)
    at Generator.throw (<anonymous>)
    at step (/usr/local/lib/node_modules/exp/node_modules/xdl/build/Project.js:1710:191)
    at /usr/local/lib/node_modules/exp/node_modules/xdl/build/Project.js:1710:402
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
```

**Comment by brentvatne** [comment]
which version of exp do you have installed?

**Comment by dooboolab** [comment]
![image](https://user-images.githubusercontent.com/27461460/35543176-12f7637e-05a7-11e8-9c0e-b55002241c1a.png)
Recent one!

**Comment by dooboolab** [comment]
exp --version
`49.0.3`

**Comment by brentvatne** [comment]
can you run `exp diagnostics` and paste the output here?

**Comment by dooboolab** [comment]
@brentvatne 
yes sure.
```
[exp] Generating diagnostics report...
[exp] You can join our slack here: https://slack.expo.io/.

Environment:
  OS: macOS High Sierra 10.13.2
  Node: 8.3.0
  Yarn: Not Found
  npm: 4.6.1
  Watchman: 4.9.0
  Xcode: Xcode 9.2 Build version 9C40b
  Android Studio: 3.0 AI-171.4443003

Diagnostics report:
  https://exp-xde-diagnostics.s3.amazonaws.com/dooboolab-6572a6eb-2012-4d7b-839d-a8147dc04ae6.tar.gz
```

**Comment by brentvatne** [comment]
are no warnings output before the `Publishing` line?

**Comment by dooboolab** [comment]
@brentvatne 
What do you mean by before `Publishing` line?
Hm.. I've given you all I can see in my screen.

**Comment by brentvatne** [comment]
ok just wanted to see if there were warnings before. can you share your project with me privately?

**Comment by dooboolab** [comment]
@brentvatne 
I am worried about my project to be shared because it is from my company tough..

**Comment by brentvatne** [comment]
@dozoisch - gotcha. @fson will investigate

**Comment by dooboolab** [comment]
@brentvatne 
Thank you. I've tried today also and still having problem. Hope it is solved soon!

Thank you again.

**Comment by allanleonardjr** [comment]
Yesterday I was having, what appears to be, the exact same issue with 49.0.3. I ended up installing a previous version of exp to test and it worked. `npm install -g exp@48.0.0`

**Comment by fson** [comment]
Hi, to help us track this error down, I've released exp v49.0.6 with better logging for this error. Can you install the latest version (`npm install -g exp`) and run the publish command again with `EXPO_DEBUG=true` and let us see the new stack trace?

**Comment by fson** [comment]
Going to close this issue in favor of #1307, which appears to be about the same error. Please report any additional information to #1307. Thanks!
